airflow/lineage/backend.py

```diff
@@ -0,0 +1,47 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Sends lineage metadata to a backend"""
+from typing import Optional
+
+# pylint: disable=unused-import
+import airflow
+
+
+class LineageBackend:
+    """Sends lineage metadata to a backend"""
+
+    def send_lineage(
+        self,
+        operator: 'airflow.models.baseoperator.BaseOperator',
```

Comment:
```suggestion
        operator: 'airflow.models.baseoperator.BaseOperator',
```

Probably we can import this at the top level. If it creates a cyclic import we can consider:
```
if TYPE_CHECKING:
    from airflow.models.baseoperator import BaseOperator  # pylint: disable=cyclic-import
```

Reply:
Ah! It does create a cyclic import and that's why I put the whole name instead of importing, didn't know the `TYPE_CHECKING` trick. I'll change it.